### PR TITLE
Improve Admin API

### LINKED ISSUE #1228: Apply the new naming/concepts to the Meta REST API

### PR DESCRIPTION
**Pull request by slinkydeveloper**
* Make sure the admin Rest API is aligned with the new terminology. Fix #1228
* Make sure error codes are aligned with the new terminology, and remove error codes we don't use anymore
* Refactor the restate_schemas_impl error type, now it tries to be more verbose whenever possible, to improve the admin experience

### CODE REVIEW

File: crates/admin/src/rest_api/invocations.rs [hunk 1]
```diff
@@ -38,8 +38,8 @@ pub struct DeleteInvocationParams {
 #[openapi(
     summary = "Terminate an invocation",
     description = "Terminate the given invocation. By default, an invocation is terminated by gracefully \
-    cancelling it. This ensures service state consistency. Alternatively, an invocation can be killed which \
-    does not guarantee consistency for service instance state, in-flight invocation to other services, etc.",
+    cancelling it. This ensures virtual object state consistency. Alternatively, an invocation can be killed which \
+    does not guarantee consistency for virtual object instance state, in-flight invocation to other services, etc.",
```
Comment (tillrohrmann) [inline crates/admin/src/rest_api/invocations.rs:42]: nit: in-flight invocations

File: crates/meta-rest-model/src/subscriptions.rs [hunk 2]
```diff
@@ -23,20 +23,19 @@ pub use restate_schema_api::subscription::{ListSubscriptionFilter, Subscription}
...
+    /// * `kafka://<cluster_name>/<topic_name>`, e.g. `kafka://my-cluster/my-topic`
     #[serde_as(as = "serde_with::DisplayFromStr")]
     #[cfg_attr(feature = "schema", schemars(with = "String"))]
     pub source: Uri,
     /// # Sink
     ///
     /// Sink uri. Accepted forms:
     ///
-    /// * `service://<service_name>/<method_name>`, e.g. `service://com.example.MySvc/MyMethod`
+    /// * `component://<component_name>/<component_name>`, e.g. `component://MySvc/MyMethod`
```
Comment (tillrohrmann) [inline crates/meta-rest-model/src/subscriptions.rs:38]: `component://<component_name>/<handler_name>`, e.g. `component://MyComp/MyHandler`?

File: crates/errors/src/lib.rs [hunk 3]
```diff
@@ -35,8 +35,8 @@ mod helper;
 // META are meta related errors.
 
 declare_restate_error_codes!(
-    RT0001, RT0002, RT0003, RT0004, RT0005, RT0006, RT0007, RT0008, RT0009, META0001, META0002,
-    META0003, META0004, META0005, META0006, META0007, META0008, META0009, META0010, META0011,
+    RT0001, RT0002, RT0003, RT0004, RT0005, RT0006, RT0007, RT0008, RT0009, META0003, META0004,
+    META0005, META0006, META0009, META0010, META0011,
```
Comment (tillrohrmann) [inline crates/errors/src/lib.rs:39]: What are we doing with retired error codes? Will they be reserved and cannot be used for future errors?

File: crates/node/src/roles/worker.rs [hunk 4]
```diff
@@ -279,7 +279,9 @@ where
 {
     // hack to suppress repeated logging of schema registrations
     // todo: Fix it
-    tracing::subscriber::with_default(NoSubscriber::new(), || schemas.overwrite(schema_updates))?;
+    let _ = tracing::subscriber::with_default(NoSubscriber::new(), || {
+        schemas.overwrite(schema_updates)
+    });
```
Comment (tillrohrmann) [inline crates/node/src/roles/worker.rs:284]: Why is it ok to swallow potential overwrite errors here?

File: crates/schema-impl/src/lib.rs [hunk 5]
```diff
@@ -221,9 +200,9 @@ impl Schemas {
     pub fn apply_updates(
         &self,
         updates: impl IntoIterator<Item = SchemasUpdateCommand>,
-    ) -> Result<(), SchemasUpdateError> {
+    ) -> Result<(), Error> {
```
Comment (tillrohrmann) [inline crates/schema-impl/src/lib.rs:203]: Why is this method returning an `Error`?

File: crates/schema-impl/src/lib.rs [hunk 6]
```diff
@@ -237,9 +216,9 @@ impl Schemas {
     pub fn overwrite(
         &self,
         updates: impl IntoIterator<Item = SchemasUpdateCommand>,
-    ) -> Result<(), SchemasUpdateError> {
+    ) -> Result<(), Error> {
```
Comment (tillrohrmann) [inline crates/schema-impl/src/lib.rs:219]: Why is this method returning an `Error`?

File: crates/schema-impl/src/error.rs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    #[error("already exists in the schema registry")]
+    #[code(unknown)]
+    Override,
+
+    // Specific resources errors
+    #[error(transparent)]
+    Component(
+        #[from]
+        #[code]
+        ComponentErrorKind,
```
Comment (tillrohrmann) [inline crates/schema-impl/src/error.rs:44]: Could that just be the `ComponentError`? Saving some characters?

File: crates/schema-impl/src/error.rs [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    ),
+    #[error(transparent)]
+    Deployment(
+        #[from]
+        #[code]
+        DeploymentErrorKind,
```
Comment (tillrohrmann) [inline crates/schema-impl/src/error.rs:50]: Same here.

File: crates/schema-impl/src/error.rs [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    ),
+    #[error(transparent)]
+    Subscription(
+        #[from]
+        #[code]
+        SubscriptionErrorKind,
```
Comment (tillrohrmann) [inline crates/schema-impl/src/error.rs:56]: And here.

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @slinkydeveloper. I think the changes look good. I left a few comments where it seemed that we could pull the clean up a bit further. +1 for merging after resolving the comments.

File: crates/schema-impl/src/lib.rs (hunk 5 above)
Comment (slinkydeveloper) [reply crates/schema-impl/src/lib.rs:203]: Sorry this is a leftover, i removed it.

File: crates/schema-impl/src/lib.rs (hunk 6 above)
Comment (slinkydeveloper) [reply crates/schema-impl/src/lib.rs:219]: Sorry this is a leftover, i removed it.

**Pull request merged**